python/tvm/contrib/hexagon/session.py

```diff
@@ -194,33 +196,36 @@ def get_graph_executor(
         self._set_device_type(graph_mod)
         return tvm.contrib.graph_executor.create(graph_json, graph_mod, self.device)
 
-    def get_aot_executor(
+    def get_graph_debug_executor(
```

Comment:
Unfortunately github didn't present this correctly. I removed `get_aot_executor` function because it didn't work correctly and also it was not used anywhere in the codebase. Therefore it will remain broken if we keep it and others might try to use it.

Reply:
Also I changed `get_graph_debug_executor` to work with the current approach. Just realized we are missing a test for that as well. I suggest to add a test for it since this api is useful for debugging.

Reply:
I think we should add a test for the `get_aot_executor` function, rather than removing it entirely.  The current tests use `get_executor_from_factory`, which calls into `_aot_executor_from_factory`.  This is exactly what we need for CI, where each model is uploaded once and run once, but isn't as useful when a model is uploaded once and run many times, potentially across multiple executions.

The `get_aot_executor` path is intended to support that use case, where an AOT module has been compiled and uploaded, and in each future occurrence only needs to be loaded.

Reply:
we could either add a test for get_aot_executor or rework `_aot_executor_from_factory` to use `get_aot_executor` function. I prefer the latter.